tests/func/test_data_cloud.py

```diff
@@ -39,7 +39,10 @@
 TEST_CONFIG = {
     Config.SECTION_CACHE: {},
     Config.SECTION_CORE: {Config.SECTION_CORE_REMOTE: TEST_REMOTE},
-    TEST_SECTION: {Config.SECTION_REMOTE_URL: ""},
+    TEST_SECTION: {
+        Config.SECTION_REMOTE_URL: "",
+        Config.SECTION_REMOTE_NO_TRAVERSE: False,
```

Comment:
It is `True` by default :slightly_smiling_face:

Reply:
Know, but it wasn't during the tests: `TEST_SECTION` didn't define `no_traverse`, and before adding:
```python
self.no_traverse = config.get(Config.SECTION_REMOTE_NO_TRAVERSE, True)
```
it was "False" by default. If I set it to `True`, `TestRemoteSSHMocked` failed :/

So I just made it "explicit" (I wasn't sure this was the expected behavior)

Reply:
/me `--verbose`

The error  was the following:

```
__________________________________________________________ TestRemoteSSHMocked.test ___________________________________________________________

self = <tests.func.test_data_cloud.TestRemoteSSHMocked testMethod=test>

    def test(self):
        self._ensure_should_run()
>       self._test_cloud()

tests/func/test_data_cloud.py:371:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/func/test_data_cloud.py:316: in _test_cloud
    status = self.cloud.status([info], show_checksums=True)
dvc/data_cloud.py:123: in status
    targets, jobs=jobs, remote=remote, show_checksums=show_checksums
dvc/remote/local/__init__.py:334: in status
    remote_exists = list(remote.cache_exists(md5s, jobs=jobs))
dvc/remote/ssh/__init__.py:277: in cache_exists
    ret = list(itertools.compress(checksums, in_remote))
/usr/lib/python3.7/concurrent/futures/_base.py:598: in result_iterator
    yield fs.pop().result()
/usr/lib/python3.7/concurrent/futures/_base.py:435: in result
    return self.__get_result()
/usr/lib/python3.7/concurrent/futures/_base.py:384: in __get_result
    raise self._exception
/usr/lib/python3.7/concurrent/futures/thread.py:57: in run
    result = self.fn(*self.args, **self.kwargs)
dvc/remote/ssh/__init__.py:269: in exists_with_progress
    return self.batch_exists(chunks, callback=progress_callback)
dvc/remote/ssh/__init__.py:248: in batch_exists
    channels = ssh.open_max_sftp_channels()
dvc/remote/ssh/connection.py:298: in open_max_sftp_channels
    self._sftp_channels.append(self._ssh.open_sftp())
.venv/lib/python3.7/site-packages/paramiko/client.py:556: in open_sftp
    return self._transport.open_sftp_client()
.venv/lib/python3.7/site-packages/paramiko/transport.py:1097: in open_sftp_client
    return SFTPClient.from_transport(self)
.venv/lib/python3.7/site-packages/paramiko/sftp_client.py:169: in from_transport
    chan.invoke_subsystem("sftp")
.venv/lib/python3.7/site-packages/paramiko/channel.py:72: in _check
    return func(self, *args, **kwds)
.venv/lib/python3.7/site-packages/paramiko/channel.py:283: in invoke_subsystem
    self._wait_for_event()
.venv/lib/python3.7/site-packages/paramiko/channel.py:1226: in _wait_for_event
    raise e
.venv/lib/python3.7/site-packages/paramiko/transport.py:2055: in run
    ptype, m = self.packetizer.read_message()
.venv/lib/python3.7/site-packages/paramiko/packet.py:459: in read_message
    header = self.read_all(self.__block_size_in, check_rekey=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <paramiko.packet.Packetizer object at 0x7f45f3141a50>, n = 16, check_rekey = True

    def read_all(self, n, check_rekey=False):
        """
        Read as close to N bytes as possible, blocking as long as necessary.

        :param int n: number of bytes to read
        :return: the data read, as a `str`

        :raises:
            ``EOFError`` -- if the socket was closed before all the bytes could
            be read
        """
        out = bytes()
        # handle over-reading from reading the banner line
        if len(self.__remainder) > 0:
            out = self.__remainder[:n]
            self.__remainder = self.__remainder[n:]
            n -= len(out)
        while n > 0:
            got_timeout = False
            if self.handshake_timed_out():
                raise EOFError()
            try:
                x = self.__socket.recv(n)
                if len(x) == 0:
>                   raise EOFError()
E                   EOFError

.venv/lib/python3.7/site-packages/paramiko/packet.py:303: EOFError
```